pkg/internal/kube/store.go

```diff
@@ -179,7 +186,7 @@ func (s *Store) cacheResourceMetadata(meta *informer.ObjectMeta) *CachedObjMeta
 			com.OTELResourceMeta[serviceNamespaceKey] = val
 		}
 	}
-	com.ServiceName, com.ServiceNamespace = s.serviceNameNamespaceForMetadata(meta)
+	com.ServiceName, com.ServiceNamespace = s.serviceNameNamespaceForMetadata(meta, "")
```

Comment:
Same here, the container name is empty, so when we cache the service name and namespace it will be different than the one we resolve.

Reply:
It seems to be this code is unnecessary, I remove the ServiceName and ServiceNamespace properties and the mentioned line and could still compile.